Code/SG Scottish manufacturer analysis Sep24.R

```diff
@@ -0,0 +1,106 @@
+
+HFSSFINAL23 <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/HFSSFINAL23.parquet")#Open linked data file for 2023
+prodnames <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/npm_hfss_prod_names.parquet")
+
+#Kantar_regcats <- fread("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/PHS HFSS Itemisation Breakdown amended.csv")
+#IGNORE - KEPT IN CASE NEEDED THEN WILL DELETE
+#files <- list.files(path = "2023 Purchase Data/product master", full.names = TRUE)#Create combined RFfile
+#dataframes <- lapply(files, function(f){
+  #fread(f, skip = 1, header = TRUE) %>%
+    #select(any_of(c("Product", "Product Desc", "Manufacturer", "Brand", "RST 4 Market", "RST 4 Sub Market", "RST 4 Extended", "Low Sugar/Calorie/Fat")))
+#}
+#)
+
+#rf_combined_SG <- do.call(bind_rows, dataframes)
+
+  #write_parquet(rf_combined_SG, "rf_combined_SG.parquet")
+
+#rf_combined_SG <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/rf_combined_SG.parquet")#Link manufacturer data to HFSSFINAL22_23 data file using a version of Catriona's 'rf_markets_combined.parquet file - saved separately.
+
+
+#QA - remove 47 products with missing manufacturer info
+#rf_combined_SG <- rf_combined_SG[-which(rf_combined_SG$Manufacturer == ""), ]
+
+
+#18/09/24 - START HERE
+
+#Get Catriona's file with HFSS categories
+prodnames <- read_parquet("/PHI_conf/PHSci-HFSS/Kantar analysis/Working Data/npm_hfss_prod_names.parquet")
+
+#Merge master purchase dataset for 2023 with HFSS regulation category file
+SG_dataset <- dplyr::left_join(HFSSFINAL23,prodnames, by=c("prodcode" = "PRODUCT"))
+             
+#Tidy up file
+SG_dataset <- subset(SG_dataset, select = -c(28, 57:60, 63:65, 68))#Delete duplicate variables
+SG_dataset <- SG_dataset[  ,c(1:28,53,29:44,59,45:52,54:58)] #re-order variables
+
+#rename category variable descriptors
+colnames(SG_dataset) [colnames(SG_dataset) %in% c("Area", "Market","Sector","Submarket","Extended")] <- c("area_desc", "market_desc", "mktsector_desc", "submkt_desc",  "extended_desc")
+
+SG_dataset <- SG_dataset %>% mutate_at(c('amtspent'), as.numeric)#Change amtspent to numeric
+
+
+#QA CHECK
+#2,421,692 observations of 59 variables
+
+#Remove products which are not included in the regulation categories and which are not HFSS = 1
+SG_dataset <- SG_dataset %>% drop_na(hfss_category)#Drop non-HFSS category products (left with 765,267 purchases for 2023)
+SG_dataset <- SG_dataset[SG_dataset$HFSS.x != 0, ]#Drop products with an HFSS score of 0 within the regulation categories (left with 474,844 HFSS purchases falling into the reg categories for 2023)
+
+
+#Write data file for SG analysis
+write_parquet(SG_dataset, "SG_dataset.parquet")
+
+
+########################################
+#PART 2: SYNTAX TO CREATE TABLE METRICS#
+#######################################
+
+#SUMMARY STATISTICS
+
+
+SGTableA <- SG_dataset%>%        
+  group_by(Manufacturer) %>% 
+  summarise(Spend= sum(amtspent))
```

Comment:
Remember to add ungroup if you group something. It doesn't make any visible difference to the dataframe, but it can cause various problems with other operations later if it remains grouped.

```suggestion
SGTableA <- SG_dataset%>%        
  group_by(Manufacturer) %>% 
  summarise(Spend= sum(amtspent)) %>%
  ungroup()
```